### PR TITLE
tail: add fixes to pass "tail-2/tail-c.sh" from GNU's test suite

### PR DESCRIPTION
**Pull request by jhscheer**

### CODE REVIEW

File: src/uu/tail/src/tail.rs [hunk 1]
```diff
@@ -188,8 +190,9 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
             if path.is_dir() {
                 continue;
             }
+            let md = metadata(path).unwrap();
             let mut file = File::open(&path).unwrap();
```
Comment (blyxxyz) [inline src/uu/tail/src/tail.rs:194]:
```suggestion
            let mut file = File::open(&path).unwrap();
            let md = file.metadata().unwrap();
```

File: src/uu/tail/src/tail.rs [hunk 2]
```diff
@@ -188,8 +190,9 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
...
-            if is_seekable(&mut file) {
+            if is_seekable(&mut file) && md.blocks() > 0 {
```
Comment (blyxxyz) [inline src/uu/tail/src/tail.rs:195]: This has to be guarded so the extra check only happens on Unix. Ideally the `.metadata()` call would also only happen on Unix.

**Comment by blyxxyz** [comment]
Ah, didn't notice this was still marked as draft, sorry.

File: src/uu/tail/src/tail.rs (hunk 1 above)
Comment (jhscheer) [reply src/uu/tail/src/tail.rs]: thanks

File: src/uu/tail/src/tail.rs (hunk 2 above)
Comment (jhscheer) [reply src/uu/tail/src/tail.rs]: Yes, I know, I had it on draft to fix this.

**Comment by jhscheer** [comment]
> Ah, didn't notice this was still marked as draft, sorry.

No problem, I appreciate your review.

**Comment by sylvestre** [comment]
_Congrats! The gnu test tests/tail-2/tail-c is now passing!_
Bravo!

**Pull request merged**